ql/indexes/inflationindex.cpp

```diff
@@ -70,23 +70,46 @@ namespace QuantLib {
                       Frequency frequency,
                       const Period& availabilityLag,
                       const Currency& currency,
-                      const Handle<ZeroInflationTermStructure>& zeroInflation)
-    : InflationIndex(familyName, region, revised, interpolated,
+                      const Handle<ZeroInflationTermStructure>& zeroInflation,
+                      bool forecastWhenPossible)
+        : InflationIndex(familyName, region, revised, interpolated,
                      frequency, availabilityLag, currency),
-      zeroInflation_(zeroInflation) {
+      zeroInflation_(zeroInflation),
+      forecastWhenPossible_(forecastWhenPossible) {
         registerWith(zeroInflation_);
     }
 
+    bool ZeroInflationIndex::forecastTodaysFixing(const Date& fixingDate) const {
+        return forecastWhenPossible_ && fixingDate >= zeroInflationTermStructure()->baseDate();
+    }
+
+
+    Rate ZeroInflationIndex::fixing(const Date& fixingDate) const {
+        return fixing(fixingDate, fixingDate);
+    }
+
     Rate ZeroInflationIndex::fixing(const Date& aFixingDate,
-                                    bool /*forecastTodaysFixing*/) const {
-        if (!needsForecast(aFixingDate)) {
+                                    bool forecastTodaysFixing) const {
+        return fixing(aFixingDate, aFixingDate, forecastTodaysFixing);
+    }
+
+    Rate ZeroInflationIndex::fixing(const Date& fixingDate,
+                                    const Date& referenceDate) const {
+        return fixing(fixingDate, referenceDate, forecastTodaysFixing(fixingDate));
+    }
+    
+    Rate ZeroInflationIndex::fixing(const Date& aFixingDate,
+                                    const Date& referenceDate,
+                                    bool forecastTodaysFixing) const {
+        if (!forecastTodaysFixing && !needsForecast(aFixingDate)) {
```

Comment:
What if I ask a fixing in the far past and pass `forecastTodaysFixing = true`? Won't this condition cause the code to call `forecastFixing` and fail because the fixing date is before the base date of the curve?